src/lib.rs

```diff
@@ -170,18 +187,43 @@ impl<T: Runtime> ClientBuilder<T> {
             rpc.system_properties(),
         )
         .await;
+        let metadata = metadata?;
+        let mortal_period = if let Some(period) = self.mortal_period {
+            period
+        } else {
+            match metadata.derive_mortal_period() {
```

Comment:
I had in mind to pass `mortal_period` into `derive_mortal_period` in order to check if it was valid but I have no strong opinions on it.

Reply:
Honestly it is not even clear to me what would be an invalid mortal period a user could set. I think we could check if `mortal_period <= BlockHashCount.next_power_of_two()`, but the tx will still be processed by the node with larger values.  Knowing that, it seems artificially limiting, especially if for some reason a runtime follows a different rule.

Reply:
Fair enough, I was thinking of the initial annoyance, if the `mortal_period > BlockHashCount`